samples/tab-nested-auth/nodejs/src/components/Msal.js

```diff
@@ -0,0 +1,143 @@
+                const accountFilter = {
+                    tenantId: context.user?.tenant?.id,
+                    homeAccountId: context.user?.id,
+                    loginHint: (await app.getContext()).user?.loginHint
+                };
+                const accountWithFilter = pca.getAccount(accountFilter);
+                if (accountWithFilter) {
+                    activeAccount = accountWithFilter;
+                    pca.setActiveAccount(activeAccount);
+                }
+            } catch (error) {
+                console.log(error);
+            }
+        }
+        return activeAccount;
+    }
+
+    // Function to get the token for the user
+    const getToken = async () => {
+        let activeAccount = await getActiveAccount();
+        const tokenRequest = {
+            scopes: ["User.Read"],
+            account: activeAccount || undefined,
+        };
+        return pca.acquireTokenSilent(tokenRequest)
+            .then((result) => {
+                console.log(result);
+                return result.accessToken;
+            })
+            .catch((error) => {
+                console.log(error);
+                // Try to get token via popup if silent acquisition fails
+                return pca.acquireTokenPopup(tokenRequest)
+                    .then(async (result) => {
+                        console.log(result);
+                        return result.accessToken;
+                    })
+                    .catch((error) => {
+                        console.log(error);
+                        return JSON.stringify(error);
+                    });
+            });
+    }
+
+    // Function to initiate login and token retrieval
+    const msallogin = async () => {
+        let isNAAResults = await nestedAppAuth.isNAAChannelRecommended();
+        if (isNAAResults == true) {
+            console.log("Nested app auth capabilities supported");
+            return initializePublicClient().then((_client) => {
+                return getToken().then((token) => {
+                    callApi(token);
+                });
+            });
+        } else {
+            console.log("Nested app auth capabilities not supported");
+        }
+    }
+
+    // Function to call the Microsoft Graph API with the access token
+    async function callApi(accessToken) {
+        const response = await fetch(
+            `https://graph.microsoft.com/v1.0/me`,
+            {
+                headers: { Authorization: accessToken },
+            }
+        );
+        if (response.ok) {
+            // Parse and set the user data
+            const jsonValue = await response.json();
+            const alignedJSON = JSON.stringify(jsonValue, null, 2);
+            setMeData(alignedJSON);
+            setIsLoggedIn(true);
+        } else {
+            const errorText = await response.text();
+            console.error("Microsoft Graph call failed - error text: " + errorText);
```

Comment:
Do you want any UI indication that the Graph call failed (or any other step) or is it sufficient to log those cases?

Reply:
Done